src/Elm/Syntax/Expression.elm

```diff
@@ -333,11 +333,11 @@ encodeLetBlock { declarations, expression } =
         ]
 
 
-encodeRecordUpdate : Node String -> List (Node RecordSetter) -> Value
-encodeRecordUpdate name updates =
+encodeRecordUpdate : Node String -> Node RecordSetter -> List (Node RecordSetter) -> Value
+encodeRecordUpdate name firstUpdate updates =
     JE.object
         [ ( "name", Node.encode JE.string name )
-        , ( "updates", JE.list (Node.encode encodeRecordSetter) updates )
+        , ( "updates", JE.list (Node.encode encodeRecordSetter) (firstUpdate :: updates) )
```

Comment:
I think it is better to write to a different JSON structure as well. Maybe add a JSON field `firstUpdate`? This makes the decoder a bit more simpler as well.

Reply:
I did it this way because it doesn't break existing JSON encoded data.

Reply:
Maybe this can be simplified for the next major version?

Reply:
@MartinSStewart very good that you thought about the compatibility of the JSON structure. Some may have overseen this.

Like @jfmengels notes, the next version will be a breaking change. There is already stuff on master that indicates this, and I'm working on #49 that will also break the JSON.

I think that the JSON interface compatibility should comply with the SemVer rules of the package itself. In that case I find it justified to break compatibility. What do you think?

Reply:
I think it makes sense, but it's best to warn about that in the documentation.

At the moment in `elm-review`, I cache every AST on the file-system (because parsing is very expensive), and I load them.
The cache is agnostic of the version of `elm-syntax`, so this means that potentially, the AST will become invalid if the version for `elm-syntax` gets bumped.

That said, if the decoding of the AST fails, then we fallback to simply parsing the file again. We don't lose much except for time on the initial run after `elm-syntax` has been bumped. So either I accept this trade-off, or I store the ASTs under a folder for the "current" version of `elm-syntax` (which I need to figure out, but in my case that is easy).

I mainly mention this for context, since `elm-review` will not crash and it's in practice not a big problem. But for other applications this might be a problem, as you need to account for the AST being potentially incorrect.

Let's discuss this in a different issue if you wish to, in order not to derail this discussion even more.

Reply:
I think what @jfmengels does in `elm-review` should be how everybody should use the JSON format. It is more for caching and such than a 'communication' interface. I'll create a story to make this clear in the README/docs. For now @MartinSStewart it may be wise to just create a breaking change. It keeps the code more clean.

Reply:
@stil4m I wrote a response in this issue if you have a moment https://github.com/stil4m/elm-syntax/issues/56

Reply:
Lets merge this. Then I can resolve potential conflicts with #59. #56 will be an improvement for a later stage (outside of this PR).